pkg/curl/schemes.go

```diff
@@ -52,6 +53,8 @@ type FileScheme interface {
 	// It may do so by fetching `u` and placing it in a buffer, or by
 	// returning an io.ReaderAt that fetchs the file.
 	Fetch(ctx context.Context, u *url.URL) (io.ReaderAt, error)
```

Comment:
I wouldn't have expected `Fetch` to cache in the first place and the documentation doesn't seem to say so either (it uses the phrasing `may`).

If we change the interface to an `io.Reader`, anyone who has an internal usage will have to consider it due to API breaking changes.

Reply:
The doc is clearly wrongly written in this case, I believe, as `uio.NewCachingReader` does cache in memory. 

We could, and ask people to get along. As to modify the behavior to make no-cache default for Fetch, that does not seem to answer my question: `I hope to know if it is just aesthetic, or for sth else?`. 

This can debate further, with a few "what if"s scenarios on both are needed or "that will break a/b/c" scenarios (I can elaborate). U-root simply gives people options to both than making one behavior default over another in its role as a package. Esp. when the reason to making one default over another is still unclear, the question.